doc/devguide/index.rst

```diff
@@ -0,0 +1,153 @@
+
+.. image:: workflow_fork.png
+   :height: 150
+   :width:  600
+   :alt:    Image showing how to fork a repo
+
+This fork will be an exact snapshot of the `svalinn/dagmc` repository at the time you clicked `fork`. Any new features
+that you wish to develop should be based from the develop branch of this repository, unless you know exactly what you're 
+doing. 
+
+Cloning
+---------
+You should now clone your fork of this repository to your local machine
+::
+   prompt %> git clone https://github.com/githubusername/dagmc
+   prompt %> cd dagmc
+
+It is prudent to also add the main svalinn repository as another remote source
+:: 
+   prompt %> git add remote upstream https://github.com/svalinn/dagmc
+
+Branching
+---------
+The base level of the repository contains folders for each of the supported monte carlo codes, the tools directory, and our
+amalgamated PyNE build. First, you need to checkout a new branch in which to keep your changes.
+::
+   prompt %> git checkout -b "my_feature_branch"
+
+To insert a new feature edit an existing file or add new ones as required, remember to update the 
+CMakeList.txt files as required. Your new changes need to be added, commited and pushed.
+::
+   prompt %> git add <files needed to add>
+   prompt %> git commit -m "This is a message that describes why we need these changes"
+
+Pushing
+---------------
+Now that your changes are commited, you push the changes to your remote branch in your clone of DAGMC
+::
+   prompt %> git push origin my_feature_branch
+
+Before pushing your local feature branch is the only place this changeset is stored, in order to let your
+remote repository know of these changes you have to push.
+
+Pull Requesting
+----------------
+Having succesfully pushed your changes to your remote fork, and if you immediately go to your fork on Github you should then 
+see a message offering to create a pull request with that branch to `svalinn/dagmc:develop`, if you click this message you can
+edit and submit the pull request. If you've waited a few tens of minutes between pushing and 
+going to Github you may have to manually create a pull request. Your pull request will launch our continuous integration tests and
+at some point in the near future your changes will pass all the unit tests or indeed may break the tests. Testing in progress and
+testing complted are shown at the bottom of your pull request.
+
+.. image:: github_testing.png
+   :height: 400
+   :width:  600
+   :alt:    Image showing when testing is launched.
+
+When testing passes your changes will be merged into develop.
+
+Refreshing your branch
+--------------------
+It should be noted that having had your pull request successfully integrated into the DAGMC mainline develop branch, your 
+clone and your local repositories develop branch will reflect the pre-pull request state of DAGMC. In order for your personal
+clones and repositories to be updated you must first pull the develop changes into your local clone.
```

Comment:
I'm going to dispute this one, only because on the next sub-section down the instructions then say to push your clone changes to your fork